python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1158 @@
+        "Unsqueeze": Unsqueeze,
+        "Gelu": Gelu,
+        "BiasGelu": BiasGelu,
+        "Where": Where,
+        "Clip": Clip,
+        "Equal": Equal,
+        "Shape": Shape,
+        "Not": Not,
+        "Tanh": Tanh,
+        "Sqrt": Sqrt,
+        "Relu": Relu,
+        "Conv": relay.frontend.onnx.Conv,
+        "Pow": Pow,
+        "Erf": Erf,
+        "CumSum": CumSum,
+        "Squeeze": Squeeze,
+        "Constant": Constant,
+        "Sub": Sub,
+        "LayerNormalization": relay.frontend.onnx.LayerNormalization,
+        "SkipLayerNormalization": relay.frontend.onnx.SkipLayerNormalization,
+        "EmbedLayerNormalization": relay.frontend.onnx.EmbedLayerNormalization,
+        # defs/reduction
+        "ReduceMax": relay.frontend.onnx.ReduceMax,
+        "ReduceMin": relay.frontend.onnx.ReduceMin,
+        "ReduceSum": relay.frontend.onnx.ReduceSum,
+        "ReduceMean": relay.frontend.onnx.ReduceMean,
+        "ReduceProd": relay.frontend.onnx.ReduceProd,
+        "ReduceLogSumExp": relay.frontend.onnx.ReduceLogSumExp,
+        "ReduceLogSum": relay.frontend.onnx.ReduceLogSum,
+        "ReduceSumSquare": relay.frontend.onnx.ReduceSumSquare,
+        "ReduceL1": relay.frontend.onnx.ReduceL1,
+        "ReduceL2": relay.frontend.onnx.ReduceL2,
+        "Expand": Expand,
+        "ConstantOfShape": ConstantOfShape,
+        "Slice": Slice,
+        "Attention": relay.frontend.onnx.Attention,
+        "Pad": Pad,
+        "Split": Split,
+        "Tile": Tile,
+        "BatchNormalization": relay.frontend.onnx.BatchNorm,
+        "GlobalAveragePool": relay.frontend.onnx.GlobalAveragePool,
+        "Flatten": relay.frontend.onnx.Flatten,
+        "MaxPool": relay.frontend.onnx.MaxPool,
+    }
+
+
+class GraphProto:
+    """A helper class for handling Relax expression copying from pb2.GraphProto.
+    Definition: https://github.com/onnx/onnx/blob/main/onnx/onnx.proto
+        Parameters
+    ----------
+    shape : dict of str to tuple, optional
+        The input shape to the graph
+    dtype : str or dict of str to str
+        The input types to the graph
+    """
+
+    current = None
+
+    def __init__(self, shape, dtype, target):
+        self._nodes = {}
+        self._inputs = {}
+        self._num_input = 0
+        self._shape = shape.copy() if shape else {}
+        self._input_names = []
+        self._dtype = dtype
+        self.opset = None
+        self._target = target
+        self.bb = relax.BlockBuilder()
+
+    def from_onnx(self, graph, opset) -> IRModule:
+        """Construct Relax expression from ONNX graph.
+        Onnx graph is a python protobuf object.
+        The companion parameters will be handled automatically.
+        However, the input names from onnx graph is vague, mixing inputs and
+        network weights/bias such as "1", "2"...
+        For convenience, we rename the `real` input names to "input_0",
+        "input_1"... And renaming parameters to "param_0", "param_1"...
```

Comment:
Where does this sanitization take place? It should probably be in some factored out place. Also, that would be good to test separately. I reiterate that name sanitization has been an enormous source of pain in the Relay importers, so it should be something we approach carefully in Relax.

Reply:
Great point, fortunately i think this comment is a relic of the past that got copy pasted in. We dont do any name mangling for inputs or parameters, we just use whatever Onnx named them.

Reply:
I see. Unfortunately, I think sanitization will be necessary or else parser roundtripping will fail for imported programs. The pretty-printer produces TVMScript, so any variable names that don't suit Python's parser will break.

Reply:
Python naming requirements:
* Var names must start with _ or a letter
* Only alphanumeric characters and underscores are permitted in var names

So the cases we would have to look out for are var names in ONNX that start with a number or (I don't know if this is possible) a var name that is the empty string

Reply:
I have now added name sanitization to the graph